cache_control.go

```diff
@@ -12,18 +13,26 @@ import (
 // underlying server's method implementation to allow control over the
 // duration and nature of caching on a per-request basis.
 type CacheControl struct {
+	sync.RWMutex
+
 	// MaxAge is maximum duration (since the original retrieval) that
 	// an item is considered fresh.
 	MaxAge time.Duration
 }
 
 func (cc *CacheControl) cacheable() bool {
-	return cc.MaxAge > 0
+	cc.RLock()
```

Comment:
I usually prefer no defer for perf (at least in cases where it is unlikely for a panic to occur). But I actually agree with you, and will switch my general approach to using defer unless there is a _proven_ perf issue.